.circleci/config.yml

```diff
@@ -0,0 +1,113 @@
+version: 2.1
+
+executors:
+  my-executor:
+    docker:
+      - image: continuumio/miniconda3
+      # astroconda/python:3.6.1
+    working_directory: ~/repo
+
+jobs:
+  build:
+    executor: my-executor
+    environment:
+      - SOURCE_BRANCH: master
+      - TARGET_BRANCH: gh-pages
+    steps:
+      - add_ssh_keys
+
+      - checkout
+
+      - restore_cache:
+          key: v1-miniconda-{{ .Branch }}
+
+      - run:
+          name: Install miniconda
+          command: |
+            ./.circleci/install_conda.sh
+
+      - save_cache:
+          key: v1-miniconda-{{ .Branch }}
+          paths:
+            - /home/circleci/miniconda
+
+      - restore_cache:
+          key: v1-miniconda-depends-{{ .Branch }}
+
+      - run:
+          name: Setup Environment
+          command: |
+            ./.circleci/setup_env.sh
+
+      - save_cache:
+          key: v1-miniconda-depends-{{ .Branch }}
+          paths:
+            - /home/circleci/miniconda
+
+      - run:
+          name: Run/convert notebooks
+          command: |
+            source activate /home/circleci/miniconda/envs/notebooks_env
+            conda info --envs
+            python ./convert.py
+
+      - run:
+          name: Check notebooks
+          command: |
+            COMMIT_RANGE=$(echo "${CIRCLE_COMPARE_URL}" | cut -d/ -f7)
+            if [[ $COMMIT_RANGE != *"..."* ]]; then
+              COMMIT_RANGE="${COMMIT_RANGE}...${COMMIT_RANGE}"
+            fi
+            source activate /home/circleci/miniconda/envs/notebooks_env
+            conda info --envs
+            python -m "nbpages.check_nbs" --commit-range=${COMMIT_RANGE}
+
+      # clone gh-pages branch, copy over files from build, commit and push
+      - deploy:
+          name: Deploy
+          command: |
+            git config --global user.email $GH_EMAIL
+            git config --global user.name $GH_NAME
+
+            git clone $CIRCLE_REPOSITORY_URL /out
+            cd /out
+
+            git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+
+            cp -aR ~/repo/* .
+
+            git add .
+            git commit -m "Automated deployment to GitHub Pages: ${CIRCLE_SHA1}" --allow-empty
+            git push origin $TARGET_BRANCH
+
+      - store_artifacts:
+          path: /out/notebooks
+
+      # - persist_to_workspace:
```

Comment:
(if this isn't needed in the end, the commented parts should be removed so they don't confuse future viewers)